src/Model.php

```diff
@@ -1592,22 +1561,24 @@ public function loadBy(string $field_name, $value)
         $system = $field->system;
         $default = $field->default;
 
+        $scope = clone $this->scope;
+
         // add condition and load record
         $this->addCondition($field_name, $value);
 
         try {
             $this->loadAny();
         } catch (\Exception $e) {
             // restore
-            array_pop($this->conditions);
+            $this->scope = $scope;
             $field->system = $system;
             $field->default = $default;
 
             throw $e;
         }
 
         // restore
-        array_pop($this->conditions);
+        $this->scope = $scope;
```

Comment:
This is not a restore, not sure what @romaninsh likes on this. Absolute no go from me!

Reply:
I will change it but where do you see the problem in ising cloned scope as replacement?

Reply:
see Discord example. Also, with deep Scopes I expect cloning to be expensive op

Reply:
@georgehristov - were you able to look into this?

Reply:
Yes, forgot about it but I will fix that

Reply:
@mvorisek better now?

Reply:
bettern but any restore needs to be done in `try/finally` fashion